### ISSUE TITLE
If the age data come together in time, steps to possibly re-run all models

### ISSUE THREAD

**Issue by andrew-edwards** [cgrandin/hake-assessment#492]
IF we do decide it's feasible and worthwhile to re-run all the models, the existing work we've already done will still help. 

All model runs are documented on https://docs.google.com/spreadsheets/d/1nhrd5NAOOVYqXC0fxqNVf-EMesIpd71oIePBBi7cB8I/edit#gid=823432831  and   https://docs.google.com/spreadsheets/d/17kSbmyr85OT0gFbrh7MtXfuTEQSeivA1Gu4iTJvFsvk/edit#gid=0 
(I've already put this table into the document, and added two rows for the 2018 models for clarity). 

Bascially, with the new ageing data the new base model will be called `2019_03_00_base_model`, which will be `2019.02.00_base_model` with the extra age data, updated average weights-at-age, updated fecundity vector, updated years for things like recruitment deviations (a lot of 2017's would normally get changed to 2018, but we didn't because there were no new age data). Changing the setting seems a bit fiddly, but obviously is what's done each year. The updated files should obviously be triple-checked as we don't have the luxury of detecting mistakes later.

All the sensitivities and bridging runs we have already are `2019.02.XX`, and so would become `2019.03.XX`. In the main google sheet I added three extra columns for each model run:
```
hake_data.ss changes (from base, or previous run for bridging models)? | hake_control.ss changes? | wtatage.ss changes?
```
with a yes or no in each column for each model. (Partly done in anticipation of the age data being available and us having to re-run everything). So this says which files have to be changed from the base model (or from the previous run for the bridging models). So if data/control/wtatage file does not change from the new base model, we just copy that into the new `2019.03.XX` directory. If it does change we change it in the same we as we did between `2019.02.00_base_model` and `2019.02.XX`. This is what we did for the runs already, using difftool to compare them.

Bit tedious, but I think it took us a day or so to redo them (we did that twice I think). Suggest we have a second person independently check each run set up (add a second name into the 'Analyst' column). 

`2019.02.19_USAgeComp` is there as a placeholder (so we wouldn't forget when redoing  everything if/when the US data are available). 

Then we have to change a bunch of `02` to `03` in the `model-setup.R`, and re-build everything.

So basically, do not change any run numbers (some of the figure captions are  bit hardwired). Add new ones if necessary. 

I think Chris and I were not keen on redoing all the runs partly because we'd already done them twice, but the way we have set up that work contributes to redoing them, plus we have a team now and can partition the work out. Plus, we now understand this stuff a bit better, which has been good. 

So, this is a possibility if we have time. But we haven't promised anyone that we can do this. Things will still break, and the MCMC's took 50 hours on our machines (24,000,000).

**Comment by aaronmberger** [comment]
Keeping track of progress on this issue:  Kelli and I have confirmed a "full" US data extraction this morning.  We need to make a decision on how to proceed as we have updated US catch data and age comp data.  kelli is working on updating the catch based .csv files on hake-assessment\data and I am working on updating those comp based .csv files and combining the US and CAN comp data.  We will then proceed to wt-at-age.  However, if we push those updates, then that changes the data used in the current catch-only built assessment.  Thoughts on how we should proceed?

**Comment by andrew-edwards** [comment]
Changes to the files in `hake-assessment/data` are used for just building the document. So that's fine.

The SS files have to be manually built by you (we don't have the code for merging Can and US data). So when you create the new `2019.03.00_base_model` directory put it in here:

https://drive.google.com/drive/folders/1BpW3dmhrnKH4D3h_66kkwxKFrQ7Bbcye

(we just made the 2019.03_models folder). We won't touch the knitr code to refer to this until the MCMCs are run. Remember to make a folder called `2019.03.00_base_model/mcmc/` in which you put the MCMC results, and then re-run ss3 for the MLEs. (Think is documented on the main GitHub readme).

**Comment by andrew-edwards** [comment]
To calculate fecundities, look in `utilities.R` at the function:
```
fec.wtatage
```
This makes it easy to calculate the fecundity by copying the wtatage matrix to the clipboard (directly from `wtatage.ss`). 

We also tried calculating average weights-at-age using `avg.wtatage()` (in same file), but then realised it's a weighted mean and so haven't used it. (For 2018 weights-at-age we had the 2015-2017 calculated last year so we could use those).

**Comment by aaronmberger** [comment]
So for fecundity we decided on having time varying values from 1975-2018 based on empirical wt-at-age data, but pre 1975 and post 2018 would be based on the long term average, correct?

**Comment by andrew-edwards** [comment]
Not quite. Time-varying for 1975-2018, pre-1975 based on 1975-2018 average, but post 2019 based on mean of 2016-2018.

See https://docs.google.com/spreadsheets/d/17kSbmyr85OT0gFbrh7MtXfuTEQSeivA1Gu4iTJvFsvk/edit#gid=0

That currently says 2018+ is mean 2015-2017 because we didn't have 2018.

We did 2015-2017 based on what was done for last year's alternative model (page 23 of 2018 assessment, third bullet). But we didn't set pre-1975 to be 1975-1979 (like last year's alternative model, second bullet on p23) as agreed at the JTC meeting, based on your analyses. I've written the text already, based on #469, where we kept track of what we did. 

Sensitivity runs (and colour heatmap plots) are already set up to look at using different assumptions. 

Make sense?

**Comment by aaronmberger** [comment]
I think so. Working on it now.

**Comment by andrew-edwards** [comment]
When you get to making a new base model, update the control etc. files to the Google drive and we can take a look (before starting MCMCs).

We did lots of 'diff-ing' between data/control/wtatatage files over the last few weeks!

Probably okay to call it `2019.03.00_base_model` as we suggested; if there are minor edits we can just update and ensure everyone re-downloads.

**Comment by aaronmberger** [comment]
Just completed 2019.03.00 and added details to the Readme_2019 file on the google drive in the models folder.   We should discuss at some point tomorrow how we want to proceed.  I also added possible sensitivities that would be straightforward to do and cover the "key" sensitivities requested by the SRG. There are heaps of others we could do but I'm thinking we just focus on these IF we decide to go full force with 2019.03.00, as opposed to just adding an executive summary in the appendix.

**Comment by aaronmberger** [comment]
Sent off MCMC run on my remote desktop - if current speed holds should take ~ 38 hours (or Thursday noon'ish).

**Comment by kellijohnson** [comment]
I wrote R code to copy the base model and generate new folders with altered control files for the first 6 sensitivity runs that are listed for 2019.03.00. I will upload the code shortly.

**Comment by aaronmberger** [comment]
Chris and Andy, are you available for a call this morning or early afternoon?  Kelli and are have another meeting until 10:30 or 11.  Would 11 this morning work (on Chris' line)?  If not, later is wide open just suggest the earliest convenient time that works for you guys.  I think we are at the point where we can finalize our decision about how to proceed with what model should be in what part of the document and sounds like from discussions there is reason to believe that we could complete the assessment document using the "full" update model (2019.03.XX models).

**Comment by cgrandin** [comment]
Yes. 11 works for me

**Comment by andrew-edwards** [comment]
I have to pop home around 11-ish for a table that's arriving. Hopefully it's 11am exactly then I can be back for 11:30. I'll have to update you. But, yes, talk would be good.

Aaron - you mentioned sensitivities. We already ran all of them (except one that needed age data) based from 2019.02.00_base_model, so if we are going to do them then we just repeat them. No decisions need to really be made, and the document is set up for them already (plus new heatmaps for different weight-at-age assumptions).

Sounds like Kelli's code will do it automatically (nice) which will save time.

**Comment by aaronmberger** [comment]
That is a discussion point I think because if we use 2019.03.00 as the model in the full document (not decided yet), do we then spend time redoing, and more importantly writing to, all the sensitivities or just a select number given time.  The ones I listed are the "key" sensitivities requested by the SRG, plus the alternative data weighting approaches which was a discussion at last years SRG (though could even ditch those if we want to be minimal).  Anyways, talk later at 11:30 then, right?

**Comment by kellijohnson** [comment]
Can we close this?

**Comment by andrew-edwards** [comment]
Yes - did you upload the code to do sensitivities? (Or at least do so after we submit). That would be useful....

**Comment by kellijohnson** [comment]
Code will be uploaded as 2b72aa8fa21dd3c3578c139cc0a260f1d986ee87
